### PR TITLE
Handle Nan-s/errors in the evaluator.

### PR DESCRIPTION
**Pull request by nick863**
# Description

Add the unit test checking thet the Errors in the evaluator are handled properly.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 1]
```diff
@@ -376,14 +376,17 @@ def evaluate(
...
 
-        # rename output columns
-        # Assuming after removing inputs columns, all columns are output columns
-        evaluator_result_df.rename(
-            columns={
-                col: "outputs." f"{evaluator_name}.{col.replace('outputs.', '')}" for col in evaluator_result_df.columns
-            },
-            inplace=True,
-        )
+        # If we use thread pool, the columns are already renamed.
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:379]: can we remove the rename logic from code_client instead?

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py [hunk 2]
```diff
@@ -376,14 +376,17 @@ def evaluate(
...
+        if not use_thread_pool:
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:380]: I don't see the code change for the error or empty value handling for some reason.

Comment (nick863) [reply src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py:380]: We are already handling the empty values [here](https://github.com/microsoft/promptflow/blob/235cb01a43e8d354909a04349cca3421bca97a1c/src/promptflow-evals/promptflow/evals/evaluate/_code_client/code_client.py#L42). If the evaluator breaks, we just add the empty line to output data frame. Should we change this behavior?

File: src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py (hunk 1 above)
Comment (nick863) [reply src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py]: Fixed.

**Pull request merged**